pkg/skaffold/initializer/init.go

```diff
@@ -40,64 +41,99 @@ func DoInit(ctx context.Context, out io.Writer, c config.Config) error {
 		}
 	}
 
+	a, err := AnalyzeProject(c)
+	if err != nil {
+		return err
+	}
+
+	newConfig, newManifests, err := Initialize(out, c, a)
+	if err != nil || c.Analyze {
```

Comment:
this is a little bit confusing that you return error when `c.Analyze` is set. 
Note: you are checking `c.Analyze` is set in two places now.  
It would be great if we only check it once and decide the course of action. WDYT

Would it make sense to move the WriteData in the `Initialize` ?
It would align with the fact that if `c.Analyze` is true, `Initialize` write results to `out`

Reply:
I agree, it is clunky :/ I'd like to get rid of the second check if I can. I'm a bit hesitant to move the `WriteData` logic into `Initialize`, as we want to be able to call them separately in the future for the transparent init functionality

Reply:
sounds good.